### PR TITLE
Archive the npm-debug.log file on jenkins jobs

### PR DESCRIPTION
**Pull request by jzoldak**
@edx/testeng

### CODE REVIEW

File: src/main/groovy/org/edx/jenkins/dsl/JenkinsPublicConstants.groovy [hunk 1]
```diff
@@ -22,7 +22,7 @@ class JenkinsPublicConstants {
   public static final Closure JENKINS_PUBLIC_ARCHIVE_ARTIFACTS = {
     return {
         pattern("reports/**/*,test_root/log/**/*.png,test_root/log/**/*.log, test_root/log/**/hars/*.har,**/nosetests.xml," +
-                "**/TEST-*.xml"
+                "**/TEST-*.xml,npm-debug.log"
```
Comment (benpatterson) [inline src/main/groovy/org/edx/jenkins/dsl/JenkinsPublicConstants.groovy]: should we just do `*.log`?

Comment (jzoldak) [inline src/main/groovy/org/edx/jenkins/dsl/JenkinsPublicConstants.groovy]: ah, that's a good idea. I'll amend this.

**Comment by estute** [comment]
@jzoldak are you sure this log is captured on the $WORKSPACE level?

**Comment by jzoldak** [comment]
@estute I was assuming that it would, as npm writes this file to the cwd and AFAIK the paver commands that do the install are run from the workspace folder. Do you have some info that I'm missing?

**Comment by estute** [comment]
@jzoldak oh, ok. I had been poking around in a worker and did not see it, but it could have been that the job had not finished.

**Comment by jzoldak** [comment]
yeah @estute it could also be that at the default logging level, the log is only written on error.

**Pull request merged**